### ISSUE TITLE
Non-differentiable loss function for predictions of halo MAH variance

### ISSUE THREAD

**Issue by aphearin** [ArgonneCPAC/diffmah#56]
PR #52 included code that calibrated the parameters of a model predicting the average halo mass across time. [The figure here](https://github.com/ArgonneCPAC/diffmah/pull/52#issuecomment-723656588) shows the results of the calibration, which I actually did in two steps. First, I minimized a loss function prediction for cumulative mass alone (to get the black curves to line up with the black). Second, I manually tuned the parameters controlling the amount of scatter in halo histories at fixed mass. Those parameters have a very small effect on the average halo growth, and mostly control the diversity of halo mass assembly histories, so it was reasonably straightforward to manually tune the parameters this way. 

The reason I couldn't implement a proper joint fit to the mean and variance is because whenever I tried to differentiate a loss function that depended on the computation of the variance, I get NaN returned by jax.grad. [Here is the portion](https://github.com/ArgonneCPAC/diffmah/blob/fcdaedce61ae3e9a83f412910eea10894403ef7f/diffmah/bundle_history_prediction.py#L767) of code implementing the computation of the variance. I tried to protect against log-zero [with this mask](https://github.com/ArgonneCPAC/diffmah/blob/fcdaedce61ae3e9a83f412910eea10894403ef7f/diffmah/bundle_history_prediction.py#L772), and I tried to protect against negative scatter with [this simple clip](https://github.com/ArgonneCPAC/diffmah/blob/fcdaedce61ae3e9a83f412910eea10894403ef7f/diffmah/bundle_history_prediction.py#L199) in the scatter, but those protections did not resolve the non-differentiability issue. And anyway, the puzzling thing is that the returned value of the scatter is perfectly fine, its values are about what I would expect, so tuning the params by hand wasn't a problem. Only jax.grad returns NaN, but I haven't found any other problems with the behavior of the function otherwise.

For present purposes, the level of agreement of the current halo MAH model from #52 seems good enough to move forward to SFR efficiency, but we'll need to solve this technical problem since we'll want to optimize our SFR efficiency model to capture scatter in the stellar-to-halo-mass relation, and optimizing the prediction for that scatter requires computing gradients through this prediction for the variance of a bundle of histories.
